src/handlers/ps2census/ZoneMessageHandler.ts

```diff
@@ -0,0 +1,50 @@
+// This handler is responsible for requeues and building the PS2EventQueueMessage with has various helpful data included.
+
+import PS2EventQueueMessage from '../messages/PS2EventQueueMessage';
+import {MaxRetryException, PS2Event} from 'ps2census';
+import {ChannelActionsInterface, QueueMessageHandlerInterface} from '../../interfaces/QueueMessageHandlerInterface';
+import PS2AlertsInstanceInterface from '../../interfaces/PS2AlertsInstanceInterface';
+import {getLogger} from '../../logger';
+import ApplicationException from '../../exceptions/ApplicationException';
+import {PS2EventInstanceHandlerContract} from '../../interfaces/PS2EventInstanceHandlerContract';
+import {injectable} from 'inversify';
+
+@injectable()
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export default class ZoneMessageHandler<T extends PS2Event & {zone_id: string}> implements QueueMessageHandlerInterface<T>{
+    private static readonly logger = getLogger('ZoneMessageHandler');
+
+    constructor(
+        private readonly instance: PS2AlertsInstanceInterface,
+        private readonly handlers: Array<PS2EventInstanceHandlerContract<T>>,
+    ) {}
+
+    public async handle(event: T, actions: ChannelActionsInterface): Promise<void> {
+        // Send to handlers
+        try {
+            await Promise.all(
+                this.handlers.map((handler) => handler.handle(
+                    new PS2EventQueueMessage(event, this.instance),
+                )),
+            );
+            actions.ack();
+        } catch (err) {
+            if (err instanceof MaxRetryException) {
+                ZoneMessageHandler.logger.error(`Maximum Census retries reached! Err: ${err.message}`);
+                actions.nack(); // TODO: Requeue
```

Comment:
Problem is with nacks you're unable to set a delay on it. So it would have to publish to another delay queue (yay more queues) and then deadletter back into the queue. But the issue here is we'd have to create a deadletter for each queue type, unless we're able to somehow change the destination queue per event to route correctly.